lib/active_merchant/billing/gateways/adyen_checkout.rb

```diff
@@ -153,6 +162,19 @@ def add_risk_data(post, options)
         end
       end
 
+      def add_3ds_data(post, options)
+        post[:additionalData] ||= {}
+        post[:additionalData][:allow3DS2] = options[:allow3DS2] if options[:allow3DS2]
+        post[:channel] = options[:channel] if options[:channel]
+        post[:origin] = options[:origin] if options[:origin]
+        post[:browserInfo] = options[:browser_info] if options[:browser_info] # Autopopulated with `populateBrowserInfoFor3ds` service enabled
```

Comment:
Have you talked with Adyen support about this service? Perhaps we don't need to do all of that https://github.com/chargify/chargify/pull/15768/files#diff-676f6becf571acae78e6f3195c34bed8R9?

If we do, though, let's remove this comment.

Reply:
I didn't talk to them, but if I understand correctly this browser's data are passed to onChange event that we don't use, this event is never fired. I found that they are providing some utils https://github.com/Adyen/adyen-3ds2-js-utils, so I just grabbed their collectBrowserInfo implementation